src/chunked-line-reader.js

```diff
@@ -0,0 +1,99 @@
+const fs = require("fs");
+const isBinaryFile = require("isbinaryfile");
+const {Readable} = require('stream');
+const {StringDecoder} = require('string_decoder');
+
+const lastIndexOf = function(buffer, length, char) {
+  let i = length;
+  while (i--) {
+    if (buffer[i] === char) { return i; }
+  }
+  return -1;
+};
+
+// Will ensure data will be read on a line boundary. So this will always do the
+// right thing:
+//
+//   lines = []
+//   reader = new ChunkedLineReader('some/file.txt')
+//   reader.on 'data', (chunk) ->
+//     line = chunk.toString().replace(/\r?\n?$/, '')
+//     lines = lines.concat(line.split(/\r\n|\n|\r/))
+//
+// This will collect all the lines in the file, or you can process each line in
+// the data handler for more efficiency.
+module.exports =
+class ChunkedLineReader extends Readable {
+  constructor(filePath, options = {}) {
+    super();
+    this.encoding = options.encoding ?? "utf8";
```

Comment:
```suggestion
    this.encoding = options?.encoding ?? "utf8";
```

Suggestion: Optional chaining, similar level of paranoia/safety as the CoffeeScript which appears to have also used optional chaining, or whatever that's called in CoffeeScript land:

```coffee
  constructor: (@filePath, options) ->
    @encoding = options?.encoding ? "utf8"
    super()
```

In case there may be an `options` object with no `encoding` property defined on it.

**(IDK if I'd be as eagle-eyed as this through the whole PR/diff if I tried to read it all, but I did notice this, may as well point it out while I see it!)**

Reply:
Well I'll happily take whatever review I can get, but hopefully we can rely somewhat on the CI here.

Great catch in this check, I'll go ahead and apply it!